plugin.audio.tripler/resources/lib/scraper.py

```diff
@@ -0,0 +1,2244 @@
+        self._item = item
+        self._itemobj = itemobj
+        self._itemdata = itemobj['data']
+        self.textbody = textbody
+
+    @property
+    def resource_path(self):
+        card_anchor = self._item.find(class_='card__anchor')
+        if card_anchor:
+            return Scraper.resource_path_for(card_anchor.attrs['href'])
+
+    @property
+    def type(self):
+        return self.__class__.__name__.lower()
+
+    @property
+    def subscription_required(self):
+        return self._itemobj.get('subscription_required')
+
+    @property
+    def id(self):
+        return str(self._itemobj['source_id'])
+
+    @property
+    def title(self):
+        return self._itemdata['title']
+
+    @property
+    def subtitle(self):
+        return self._itemdata['subtitle']
+
+    @property
+    def _itemtime(self):
+        return time.strptime(self._itemdata['subtitle'], '%d %B %Y')
+
+    @property
+    def date(self):
+        return time.strftime(DATE_FORMAT, self._itemtime)
+
+    @property
+    def year(self):
+        return self._itemtime[0]
+
+    @property
+    def aired(self):
+        return self.date
+
+    @property
+    def duration(self):
+        duration = self._itemobj.get('data', {}).get('duration', {})
+        if not duration:
+            audio_file = self._itemdata.get('audio_file')
+            if audio_file:
+                duration = audio_file['duration']
+            else:
+                duration = 0
+        return round(duration)
+
+    @property
+    def thumbnail(self):
+        return self._itemdata['image']['path'] if 'image' in self._itemdata.keys() else ''
+
+    @property
+    def url(self):
+        audio_file = self._itemdata.get('audio_file')
+        if audio_file:
+            return audio_file['path']
+        else:
+            ts = self._itemdata['timestamp']
+            l = self.duration
+            return 'https://ondemand.rrr.org.au/getclip?bw=h&l={}&m=r&p=1&s={}'.format(l, ts)
+
+    def to_dict(self):
+        item = {
+            'type':          self.type,
+            'id':            self.id,
+            'attributes': {
+                'title':         self.title,
+                'subtitle':      self.subtitle,
+                'textbody':      self.textbody,
+                'date':          self.date,
+                'year':          self.year,
+                'aired':         self.aired,
+                'duration':      self.duration,
+                'url':           self.url,
+                'thumbnail':     self.thumbnail,
+            },
+            'links': {
+                'self': self.resource_path,
+            }
+        }
+        if self.subscription_required:
+            item['links']['subscribe'] = '/subscribe'
+        return item
+
+
+class Archive(AudioItem):
+    ''
+
+class Broadcast(AudioItem):
+    ''
+
+class Segment(AudioItem):
+    ''
+
+class Podcast(AudioItem):
+    ''
+
+
+if __name__ == "__main__":
+    stderr = True
```

Comment:
There is no use for `stderr` in Kodi. Not even for debugging. Use `xbmc.log()` if needed.

Reply:
This enables `stderr` logging only for usage outside Kodi. (I'm removing it anyway)